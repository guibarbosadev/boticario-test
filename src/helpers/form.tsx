import React from 'react';
import { Field } from 'react-final-form';
import { composeValidators } from './validators';
import { TextInput, TextInputProps } from 'react-native';

export function renderInput(
    name: string,
    inputProps: TextInputProps = {},
    validators: ((value: string) => string | undefined)[] = []
) {
    return (
        <Field
            name={name}
            validate={composeValidators(...validators)}
            render={({ input }) => (
                <TextInput
                    value={input.value}
                    onChangeText={input.onChange}
                    {...inputProps}
                />
            )}
        />
    );
}
