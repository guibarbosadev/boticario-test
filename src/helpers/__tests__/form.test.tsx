import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import { renderInput } from '../form';
import { Form } from 'react-final-form';

describe('Form | Helpers', () => {
    it('renderInput method returns as expected', () => {
        const Input = () => renderInput('some-name');
        const { toJSON } = render(
            <Form onSubmit={() => {}} render={() => <Input />} />
        );

        expect(toJSON()).toMatchSnapshot();
    });

    it('renderInput method uses input.value', () => {
        const props = { value: 'some value' };
        const Input = () => renderInput('some-name', props);
        const { getByDisplayValue } = render(
            <Form onSubmit={() => {}} render={() => <Input />} />
        );

        expect(getByDisplayValue(props.value)).toBeTruthy();
    });

    it('renderInput method uses input.placeholder', () => {
        const props = { placeholder: 'Some placeholder' };
        const Input = () => renderInput('some-name', props);
        const { queryByPlaceholder } = render(
            <Form onSubmit={() => {}} render={() => <Input />} />
        );

        expect(queryByPlaceholder(props.placeholder)).toBeTruthy();
    });

    it('input values changes when users types something', () => {
        const value = 'Some value';
        const Input = () => renderInput('some-name', { testID: 'my-input' });
        const { getByTestId, queryByDisplayValue } = render(
            <Form onSubmit={() => {}} render={() => <Input />} />
        );

        fireEvent.changeText(getByTestId('my-input'), value);
        expect(queryByDisplayValue(value)).toBeTruthy();
    });
});
