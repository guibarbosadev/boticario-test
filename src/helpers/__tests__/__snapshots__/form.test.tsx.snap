// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form | Helpers renderInput method returns as expected 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
  value=""
/>
`;
